src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4829,7 +4829,9 @@ static int get_active_processor_count() {
   return cpu_count;
 }
 
-int os::Linux::active_processor_count() {
+// Return the active CPUs irrespective of the system being containerized
+// or not. Thus, this returns the host active cpus in a container.
```

Comment:
No it doesn't! The comment for the function you call states clearly:
```
// Get the current number of available processors for this process.
```
This is not related to "physical count" at all!

Reply:
OK. Happy to change the name to something more suitable. The idea is to have a function that returns the "available" cpus of the node/host whether in container or not. How about `available_host_processor_count()`?

Reply:
Just to be clear. It's not a goal to introduce new semantics.  What we need is an public accessible way to query this information.

Reply:
It doesn't make sense to try and add an API that is exactly the same as `os::available_processors` except on Linux. This is a Linux and Linux container problem.

Reply:
OK. Doing an alternative PR then.